### ISSUE TITLE
[BUG Report]: Getting error in BatchNormalization(),

### ISSUE THREAD

**Issue by N-E-W-T-O-N** [SciSharp/TensorFlow.NET#1276]
### Description

SO i have implemented this simple mode
```
private ILayersApi layers = tf.keras.layers;
private IModel? model{get;set;}

public void BuildModel(int height, int width, int depth, int classNumber)
    {
        // Check the Diff in GlobalAveragePooling2D() & AveragePooling2D()
        // input layer
        //keras.Sequential()
        var inputs = layers.Input(shape: (height, width, depth), name: "img");  //(32, 32, 3), name: "img");

        // convolutional layer
        var x = layers.Conv2D(8, (5, 5), padding: "same", activation: "relu").Apply(inputs);
        x = layers.BatchNormalization().Apply(x);
        x = layers.MaxPooling2D(pool_size: (2, 2)).Apply(x);

        x = layers.Conv2D(16, kernel_size: (3, 3), activation: "relu", padding: "same").Apply(x);
        x = layers.BatchNormalization().Apply(x);

        x = layers.Conv2D(16, kernel_size: (3, 3), activation: "relu", padding: "same").Apply(x);
        x = layers.BatchNormalization().Apply(x);
        x = layers.MaxPooling2D(pool_size: (2, 2)).Apply(x);

        x = layers.Conv2D(32, kernel_size: (3, 3), activation: "relu", padding: "same").Apply(x);
        x = layers.BatchNormalization().Apply(x);

        x = layers.Conv2D(32, kernel_size: (3, 3), activation: "relu", padding: "same").Apply(x);
        x = layers.BatchNormalization().Apply(x);
        x = layers.MaxPooling2D(pool_size: (2, 2)).Apply(x);

        x = layers.Flatten().Apply(x);
        x = layers.Dense(128, activation: "relu").Apply(x);
        x = layers.BatchNormalization().Apply(x);
        x = layers.Dropout(0.5f).Apply(x);

        x = layers.Flatten().Apply(x);
        x = layers.Dense(128, activation: "relu").Apply(x);
        x = layers.BatchNormalization().Apply(x);
        x = layers.Dropout(0.5f).Apply(x);

        // output layer
        var outputs = layers.Dense(classNumber, "softmax").Apply(x);
        // build keras model
        model = tf.keras.Model(inputs, outputs, name: "traffic_resnet");
    }
   
```
    
    Now inBatchNormalization().Apply(x) i am getting the following error
    
```
    Unhandled exception. System.NotImplementedException
   at Tensorflow.Framework.smart_module.smart_constant_value(Tensor pred)
   at Tensorflow.Framework.smart_module.smart_cond[T](Tensor pred, Func`1 true_fn, Func`1 false_fn, String name)
   at Tensorflow.Keras.Utils.tf_utils.smart_cond[T](Tensor pred, Func`1 true_fn, Func`1 false_fn, String name)
   at Tensorflow.Keras.Layers.BatchNormalization._fused_batch_norm(Tensor inputs, Tensor training)
   at Tensorflow.Keras.Layers.BatchNormalization.Call(Tensors inputs, Tensors state, Nullable`1 training, IOptionalArgs optional_args)
   at Tensorflow.Keras.Engine.Layer.Apply(Tensors inputs, Tensors states, Nullable`1 training, IOptionalArgs optional_args)
```

Now on Close Inspection error is  originated from this function

   public static bool? smart_constant_value(Tensor pred)
{
	NDArray nDArray = tensor_util.constant_value(pred);
	if ((object)nDArray == null)
	{
		IntPtr[] result = Enumerable.ToArray(Enumerable.Select(Binding.range(pred.op.NumOutputs), (int x) => IntPtr.Zero));
		if (!c_api.TF_TryEvaluateConstant(pred.graph, pred._as_tf_output(), result, Binding.tf.Status) || c_api.TF_GetCode(Binding.tf.Status) != 0)
		{
			return null;
		}
		throw new NotImplementedException("");
	}
	return nDArray;
}

### Configuration and Other Information

 <PackageReference Include="SciSharp.TensorFlow.Redist" Version="2.16.0" />
 <PackageReference Include="TensorFlow.Keras" Version="0.15.0" />
 <PackageReference Include="TensorFlow.NET" Version="0.150.0" />
 
 OS:win 11
 Dotnet 8

**Comment by N-E-W-T-O-N** [comment]
Just realised forgot to call

graph.Exit()
